themes-default/slim/views/vue-components/root-dirs.mako

```diff
@@ -0,0 +1,146 @@
+    </div>
+</script>
+<script>
+Vue.component('root-dirs', {
+    template: '#root-dirs-template',
+    data() {
+        /* @TODO: Change these to a list of objects, e.g.: (for @p0psicles)
+         * rootDirs = [
+         *     { path: 'rootDir0Path', default: true },
+         *     { path: 'rootDir1Path' },
+         * ]
+         */
+        let rootDirs = [];
+        let defaultDir = null;
+        const rawRootDirs = MEDUSA.config.rootDirs;
+        if (rawRootDirs.length) {
+            rootDirs = rawRootDirs.slice(1);
+            defaultDir = parseInt(rawRootDirs[0], 10);
+        }
+        return {
+            rootDirs,
+            defaultDir,
+            selectedDir: defaultDir,
+        };
+    },
+    computed: {
+        rootDirText() {
+            if (this.defaultDir === null || this.rootDirs.length < 1) return '';
+            return this.defaultDir.toString() + '|' + this.rootDirs.join('|');
+        }
+    },
+    methods: {
+        markDefault(index) {
+            return (index === this.defaultDir) ? '*' : '';
+        },
+        add() {
+            // Add a new root dir
+            $(this.$el).nFileBrowser(path => {
+                if (path.length === 0) return;
+
+                this.rootDirs.push(path);
+
+                // If this the first root dir, set it as default
+                if (this.defaultDir === null) {
+                    this.defaultDir = this.selectedDir = 0;
+                }
+
+                this.saveRootDirs();
+            });
+        },
+        edit() {
+            // Edit a root dir's path
+            $(this.$el).nFileBrowser(path => {
+                if (path.length === 0) return;
+                this.rootDirs[this.selectedDir] = path;
+                // Force the component to re-render
+                this.$forceUpdate();
+
+                this.saveRootDirs();
+            },
+            {
+                initialDir: this.rootDirs[this.selectedDir]
+            });
+        },
+        remove() {
+            // Remove a root dir
+            const thisDir = this.selectedDir;
+
+            // Remove the root dir from the list
+            this.rootDirs.splice(thisDir, 1);
+
+            // Pick a new selection
+            if (this.rootDirs.length) {
+                this.selectedDir = (thisDir > 0) ? thisDir - 1 : 0;
+            } else {
+                this.selectedDir = null;
+            }
+
+            // If we deleted the current default root dir
+            if (this.defaultDir !== null && thisDir === this.defaultDir) {
+                // If there are options left then pick a new default
+                this.defaultDir = this.selectedDir;
+            }
+
+            this.saveRootDirs();
+        },
+        setDefault() {
+            if (this.selectedDir === this.defaultDir) return;
+            this.defaultDir = this.selectedDir;
+            this.saveRootDirs();
+        },
+        saveRootDirs() {
```

Comment:
@OmgImAlexis Should this update `MEDUSA.config.rootDirs` too?

Reply:
For now let's keep changes local, thinking of removing `MEDUSA` and replacing it a puex/vuex store.